lms/extensions/feature_flags/_feature_flags.py

```diff
@@ -0,0 +1,55 @@
+"""
+Feature flags class.
+
+This module exports a :class:`FeatureFlags` class with methods for adding
+feature flag providers and for querying whether a feature flag is active for a
+request.
+
+The :class:`FeatureFlags` class is for internal use only. Outside code would
+normally only access it indirectly via public methods like
+``config.add_feature_flag_provider()`` and ``request.feature()``.
+See the :mod:`~lms.extensions.feature_flags` package for documentation of this
+public interface.
+"""
+
+__all__ = ["FeatureFlags"]
+
+
+class FeatureFlags:
+    """
```

Comment:
I notice that the provider API doesn't provide a way to enumerate the set of supported feature flags. If you were to build, eg. an interface for managing feature flags stored in cookies, you would presumably want some way to do that.

Reply:
Yeah we may want that one day but I'm not sure exactly how we'd implement it. One thing I wanted to get away from is the way that h's system makes you send a PR to add a ff before you can use it, the further ones to move them into pending removals and then to remove them. I wanted to simply be able to use a feature flag, without all this hassle.

Maybe that a config file / envvar and/or DB setting can list all the ff's _that will appear in the management API_ which is a separate list that has nothing to do with what ff's the code can or does query.

Reply:
> I wanted to get away from is the way that h's system makes you send a PR to add a ff before you can use it

What prevents a FF from being added in the same PR that makes use of it? Is that because prod instances will delete (that is, turn off) any FFs they don't think exists on startup, which is a problem if the FF + code using it only exists on QA? If so, then that won't be a problem for the lms app if there are separate QA + prod databases. Are there?